include/deal.II/fe/mapping_isoparametric.h

```diff
@@ -0,0 +1,736 @@
+ * cell is at the boundary of the domain you will likely already have attached
+ * a manifold object to all cells that can then also be used by the mapping
+ * classes for higher order mappings.
+ *
+ * <h4>Behavior along curved boundaries and with different manifolds</h4>
+ *
+ * As described above, one often only knows a manifold description of a
+ * surface but not the interior of the computational domain. In such a case, a
+ * FlatManifold object will be assigned to the interior entities that
+ * describes a usual planar coordinate system where the additional points for
+ * the higher order mapping are placed exactly according to a bi-/trilinear
+ * mapping. When combined with a non-flat manifold on the boundary, for
+ * example a circle bulging into the interior of a square cell, the two
+ * manifold descriptions are in general incompatible. For example, a
+ * FlatManifold defined solely through the cell's vertices would put an
+ * interior point located at some small distance epsilon away from the
+ * boundary along a straight line and thus in general outside the concave part
+ * of a circle. If the polynomial degree of MappingQ is sufficiently high, the
+ * transformation from the reference cell to such a cell would in general
+ * contain inverted regions close to the boundary.
+ *
+ * In order to avoid this situation, this class applies an algorithm for
+ * making this transition smooth using a so-called transfinite interpolation
+ * that is essentially a linear blend between the descriptions along the
+ * surrounding entities. In the algorithm that computes additional points, the
+ * compute_mapping_support_points() method, all the entities of the cells are
+ * passed through hierarchically, starting from the lines to the quads and
+ * finally hexes. Points on objects higher up in the hierarchy are obtained
+ * from the manifold associated with that object, taking into account all the
+ * points previously computed by the manifolds associated with the
+ * lower-dimensional objects, not just the vertices. If only a line is
+ * assigned a curved boundary but the adjacent quad is on a flat manifold, the
+ * flat manifold on the quad will take the points on the deformed line into
+ * account when interpolating the position of the additional points inside the
+ * quad and thus always result in a well-defined transformation.
+ *
+ * The interpolation scheme used in this class makes sure that curved
+ * descriptions can go over to flat descriptions within a single layer of
+ * elements, maintaining the overall optimal convergence rates of the finite
+ * element interpolation. However, one does often get better solution
+ * qualities if the transition between curved boundaries and flat interior
+ * domains is spread over a larger range as the mesh is refined. This is
+ * provided by the special manifold TransfiniteInterpolationManifold.
+ */
+template <int dim, int spacedim = dim>
+class MappingIsoparametric : public Mapping<dim, spacedim>
```

Comment:
I still think the name is misleading. I see no enforcement between this Mapping and the FE you use for the discretization. One may use `MappingIsoparametric(FE_Q<dim>(2))` and `DoFHandler(FE_RaviarThomas)`. 

How about simply `MappingFE`? This is, afterall, the precursor of `MappingFEField`.

Reply:
Strictly speaking, MappingIsoparametric is used when the same FE is used to describe the geometry and the discretization. You could call this `MappingIsoparametric`, if it took a DoFHandler, and extracted the fe from the `DoFHandler`. In the current implementation, this is more generic (and I like it) but should be renamed.

One can obtain Isoparametric finite element methods by using the same FE for the this mapping and for the dofhandler, but this should not be suggested by the name, unless you enforce it.